backend/src/storage/storage.module.ts

```diff
@@ -0,0 +1,38 @@
+import { S3Client, S3ClientConfig } from "@aws-sdk/client-s3";
+import { Module } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+
+const s3ClientFactory = {
+	provide: "STORAGE_CLIENT",
+	useFactory: (configService: ConfigService): S3Client | null => {
+		const fileUpload = configService.get<boolean | "s3" | "minio">("FILE_UPLOAD");
+		if (!fileUpload) {
+			return null;
+		}
```

Comment:
It seems that the type of `fileUpload` is always `string`(`"false" | "s3" | "minio"`).